### ISSUE TITLE
Catch case of failing promises

### ISSUE THREAD

**Issue by swapnilraj** [JorikSchellekens/lyricist#2]
When the promise for getting the lyrics fail for any reason no further requests can be made.

**Comment by JorikSchellekens** [comment]
Any ideas on this now that you're working on error monitoring?

**Comment by swapnilraj** [comment]
Yes, I have been looking at the code and seeing where we usually get the errors. The two places which are impure and the parser and the provider; for the fix I am adding catches and notifying `bugsnag` with more metadata that will help fix the bugs. `bugsnag` also has a cool feature where it will catch any uncaught exceptions and notify the server with some metadata.
